Automattic/wp-calypso
# Include designType in api request to create new site

This PR replaces #17514

Currently new sites created from the domain step do not persist the designType selected in an earlier step (for those flows that include that step).

# Testing instructions

* create a new site from `/start`
* create another new site from `/start/pressable`

In both cases, using the network tab, you should see designType has been passed to the sitesNew api call (and persisted as a site option along with theme and vertical).  The fact that these attributes have been persisted can also be verified by querying wp with `get_blog_option( blog_id, 'options' )`

Also verify that other flows that do not include the design step have not been broken:

* create a site from `/start/rebrand-cities`

Verify that the flow is still working and there is no designType in api request
